File: Assets/Common/Tools/EngineExt.cs
using Assets.Common.Log;
using Assets.Common.Resource;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

namespace Assets.Common.Tools
{
    public class EngineExt
    {
        public static GameObject s_DontDestroyParent = null;

        public static bool Raycast(Ray ray, out RaycastHit hitInfo)
        {
            return Physics.Raycast(ray, out hitInfo);
        }

        public static int RandomInt(int min, int max)
        {
            return UnityEngine.Random.Range(min, max);
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/Assets/Common/Tools/EngineExt.lua
---@class CS.Assets.Common.Tools.EngineExt : CS.System.Object
CS.Assets.Common.Tools.EngineExt = {}

---@field public CS.Assets.Common.Tools.EngineExt.s_DontDestroyParent : CS.UnityEngine.GameObject
CS.Assets.Common.Tools.EngineExt.s_DontDestroyParent = nil

---@return CS.Assets.Common.Tools.EngineExt
function CS.Assets.Common.Tools.EngineExt()
end

---@param ray : CS.UnityEngine.Ray
---@param hitInfo : CS.UnityEngine.RaycastHit
---@return CS.System.Boolean
function CS.Assets.Common.Tools.EngineExt.Raycast(ray, hitInfo)
end

---@param min : CS.System.Int32
---@param max : CS.System.Int32
---@return CS.System.Int32
function CS.Assets.Common.Tools.EngineExt.RandomInt(min, max)
end